### DISCUSSION TITLE
Select a specific country/server type

### DISCUSSION THREAD

**Discussion by ThxAndBye** [bubuntux/nordlynx#8] [Ideas]
Hey,

thanks for the effort you put into the container.  
I know there is the `QUERY` variable to add to the API request, but I think it's a bit difficult to connect to a different country currently. 

Instead of adding something like `- QUERY=filters\[country_id\]=209`, would it be feasible to make country selection (and maybe also selecting the server type like ` - QUERY=filters\[servers_groups\]\[identifier\]=legacy_p2p`) a bit easier by creating a variable for this purpose?  

E.g. a `COUNTRY_ID` and `SERVER_TYPE` variable would make this a bit easier and would require less research into the NordVPN recommendation API.

**Comment by ThxAndBye** [comment]
As a reference for anyone wanting to connect to a specific country, I queried the country ids and made a list:  
(As of 18th of December 2021)

- 2, Albania
- 10, Argentina
- 13, Australia
- 14, Austria
- 21, Belgium
- 27, Bosnia and Herzegovina
- 30, Brazil
- 33, Bulgaria
- 38, Canada
- 43, Chile
- 52, Costa Rica
- 54, Croatia
- 56, Cyprus
- 57, Czech Republic
- 58, Denmark
- 68, Estonia
- 73, Finland
- 74, France
- 80, Georgia
- 81, Germany
- 84, Greece
- 97, Hong Kong
- 98, Hungary
- 99, Iceland
- 100, India
- 101, Indonesia
- 104, Ireland
- 105, Israel
- 106, Italy
- 108, Japan
- 114, South Korea
- 119, Latvia
- 125, Lithuania
- 126, Luxembourg
- 128, North Macedonia
- 131, Malaysia
- 140, Mexico
- 142, Moldova
- 153, Netherlands
- 156, New Zealand
- 163, Norway
- 174, Poland
- 175, Portugal
- 179, Romania
- 192, Serbia
- 195, Singapore
- 196, Slovakia
- 197, Slovenia
- 200, South Africa
- 202, Spain
- 208, Sweden
- 209, Switzerland
- 211, Taiwan
- 214, Thailand
- 220, Turkey
- 225, Ukraine
- 227, United Kingdom
- 228, United States
- 234, Vietnam

**Reply by ThxAndBye** [reply]
For anyone wanting to query themselves (in the future) here is the python code I wrote:  
```python
import json
import urllib.request

api_url = "https://api.nordvpn.com/v1/servers/recommendations?&limit=1&filters[country_id]="


def get_country_by_id(country_id):
    with urllib.request.urlopen(api_url + str(country_id)) as url:
        data = json.loads(url.read().decode())
        if len(data) == 1:
            country = data[0]['locations'][0]['country']
            if country['id'] == country_id:
                return country['name']


if __name__ == '__main__':
    for country_itr in range(256):
        result = get_country_by_id(country_itr)
        if result is not None:
            print(str(country_itr) + ", " + str(result))
```

**Reply by yash-fn** [reply]
faster version:
```bash
curl --silent "https://api.nordvpn.com/v1/servers/countries" | jq --raw-output '.[] | [.id, .name] | @tsv'
```

**Comment by bubuntux** [comment]
here is a good read about nordvpn api https://sleeplessbeastie.eu/2019/02/18/how-to-use-public-nordvpn-api/

**Reply by CaTeNdrE** [reply]
This is probably so obvious but I've been banging my head against it.  How do I use the QUERY/filter for a specific known hostname 
e.g. us6285.nordvpn.com  / United States #6285?

I've tried various 
```
QUERY=filters\[server\]\[hostname\]='us6285.nordvpn.com'
QUERY=filters\[hostname\]='us6285.nordvpn.com'
QUERY=filters\[server_name\]='United States #6285'
```

I'm missing something.

**Reply by CaTeNdrE** [reply]
> 

Before migrating from bubuntux/nordvpn I would use the CONNECT env var 
```
    environment:                
      - CONNECT=us6285

```

**Reply by yash-fn** [reply]
```yaml
nordvpn:
    image: bubuntux/nordlynx
    cap_add:
      - NET_ADMIN #required
    environment:
      - PRIVATE_KEY=PRIVATE_KEY
      - END_POINT=us6285.nordvpn.com:51820
      - PUBLIC_KEY=PUBLIC_KEY
    restart: unless-stopped

```

**Reply by CaTeNdrE** [reply]
> > 
> 
> Before migrating from bubuntux/nordvpn I would use the CONNECT env var
> 
> ```
>     environment:                
>       - CONNECT=us6285
> ```

Finally got it to work by specifying both the END_POINT as well as the PUBLIC_KEY as part of the docker compose environment.  

> 
> ```
>     environment:                
>       - END_POINT=us6285.nordvpn.com:51820
>       - PUBLIC_KEY=xxxxxxxxxxxxxxxxxxxxxxxxxxxxx
> ```

**Reply by bsylvain** [reply]
The article is a tutorial on jq, there is like 3 different API call. It gives nothing on how the api itself works.

**Reply by imthenachoman** [reply]
Cause Nord does not have any  public documentation about the API. That article is written by someone tryign to show folks what they know about the API.

And the article is not a tutorial on `jq`. The author is just using `jq` to clean up the output from the API.

**Reply by b-kamphorst** [reply]
For you information: a related discussion took place in https://github.com/bubuntux/nordlynx/issues/56 ([PR](https://github.com/bubuntux/nordlynx/pull/60)) -- this proposes a variable `QUERY_JQ` that allows the user to select a country with a jq query as in the referenced article.

I have a working set-up so I don't need that PR to be merged, but if this meets (or not!) a recurring need than that might be valuable information to the developers.
